src/cluster/pager.rs

```diff
@@ -103,6 +147,29 @@ impl<
   pub fn has_more(&self) -> bool {
     self.has_more_pages.unwrap_or(false)
   }
+
+
+  /// Unsigned byte is characterized by one byte or 8 bits.
+  /// Each unsigned byte can represent a number from 0 to 255,
+  /// so we can safely pack the state into 3 chars per byte.
+  pub fn pager_state(&self) -> Option<String> {
```

Comment:
Please consider creating following structure as a pager state:

_pager.rs_
```rust
#[derive(Clone, PartialEq, Debug)]
pub struct PagerState {
  cursor: Option<CBytes>,
  has_more_pages:  Option<bool>
}

impl PagerState {
  pub fn has_more(&self) -> bool {
    self.has_more_pages.unwrap_or(false)
  }

  pub fn get_cursor(&self) -> Option<CBytes> {
    self.cursor.clone()
  }
}
```
Since we are going to provide state ejection functionality we need to get more strict about what we call a pager state. Previously we didn't pay much attention on that and we used **Cassandra** pager state (`Option<CBytes>`). Naturally, **Driver** pager state should include both `cursor` and `has_more_pages`.

After implementing this structure please change `QueryPager` so it had a form:

```rust
pub struct QueryPager<'a, Q: ToString, P: 'a> {
  pager: P,
  pager_state: PagerState,
  query: Q,
}
```
and consume state values where needed.

The same is applicable for `ExecPager`

Reply:
I have performed the refactor and I have followed your suggestion. Thank you.